src/flamegraph.cpp

```diff
@@ -274,10 +274,83 @@ QBrush brushImpl(uint hash, BrushType type)
     return brushes.at(hash % brushes.size());
 }
 
-template<typename T>
-QBrush brush(const T& entry, BrushType type)
+QBrush brushBinary(const Data::Symbol& symbol)
+{
+    static QHash<QString, QBrush> brushes;
+
+    if (brushes.contains(symbol.binary)) {
+        return brushes[symbol.binary];
+    }
+
+    QBrush _brush(brushImpl(qHash(symbol.symbol), BrushType::Hot));
+    brushes[symbol.binary] = _brush;
+    return _brush;
+}
+
+QBrush brushKernel(const Data::Symbol& symbol)
+{
+    static QBrush kernel(QColor(255, 0, 0, 125));
+    static QBrush user(QColor(0, 0, 255, 125));
+
+    if (symbol.isKernel) {
+        return kernel;
+    }
+    return user;
+}
+
+bool isSystemPath(const QString& path)
 {
-    return brushImpl(qHash(entry), type);
+    const auto sysPaths = Settings::instance()->systemPaths();
+    for (const auto& sysPath : sysPaths) {
+        if (path.startsWith(sysPath)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool isUserPath(const QString& path)
+{
+    const auto userPaths = Settings::instance()->userPaths();
+    for (const auto& userPath : userPaths) {
+        if (path.startsWith(userPath)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+QBrush brushSystem(const Data::Symbol& symbol)
+{
+    static QBrush system(QColor(0, 125, 0, 125));
+    static QBrush user(QColor(200, 200, 0, 125));
+    static QBrush unkown(QColor(50, 50, 50, 125));
+
+    // I have seen [ only on kernel calls, but am not sure if this is correct
+    if (symbol.path.isEmpty() || symbol.path.startsWith(QLatin1String("["))) {
+        return unkown;
+    } else if (isSystemPath(symbol.path)) {
+        if (isUserPath(symbol.path)) {
+            return user;
+        }
+        return system;
+    }
+    return user;
+}
+
+template<typename T>
+QBrush brush(const T& entry, Settings::ColorScheme scheme)
+{
+    switch (scheme) {
+    case Settings::ColorScheme::Binary:
+        return brushBinary(entry);
+    case Settings::ColorScheme::Kernel:
+        return brushKernel(entry);
+    case Settings::ColorScheme::System:
+        return brushSystem(entry);
+    default:
```

Comment:
don't use `default`, explicitly list all options and add a `Q_UNREACHABLE()` below the switch - this way we get compiler warnings when a new enumerator is added and we forgot to add it here